### ISSUE TITLE
GObservation::npred(GModel&): Gradient vector

### ISSUE THREAD

**Issue by sergiohcdna** [sergiohcdna/ctaAnalysis#7]
> Hi!
> 
> Yeah, I thought I updated the instructions to run the notebook. I am going to update the instructions. You can create the fits file using the script `create_dmtable.py` under the examples folder.
> 
> To check all the options you can use `$ python create_dmtable.py --help`
> 
> To create the spectrum file used for the TCAP DM session you can use:
> 
> `$ python create_dmtable.py --srcname AquariusII --jfactor 1.8621e+18 --sigmav 3.0e-26 --z 0 --mpoints 100 --emin 30.0 --nebins 450`
> 
> Let me know if this work for you!

Hi,

I'm following along with this issue and got a seperate error to the user above after creating the .fits file using create)dmtable.py .

ERROR in GObservation::npred(GModel&): Gradient vector size 1 mismatches number 0 of model parameters.

I can't find any relevant gammalib documentation on this error. Any help on how to resolve this would be much appreciated.

Kind regards

_Originally posted by @kjrule in https://github.com/sergiohcdna/ctaAnalysis/issues/6#issuecomment-917552186_

**Comment by sergiohcdna** [comment]
Hi @kjrule

Can you upload the exact command you are using?

**Comment by sergiohcdna** [comment]
Also, if you can upload the exact error you get, that will helps a lot :)

**Comment by kjrule** [comment]
Hi,
I am following through the get_dmulimits notebook. I ran all of the other notebooks in the repository and had an error that I was missing dmspectrum.fits so I created the fits file using

$ python create_dmtable.py --srcname AquariusII --jfactor 1.8621e+18 --sigmav 3.0e-26 --z 0 --mpoints 100 --emin 30.0 --nebins 450

and moved it into the notebook directory and named it dmspectrum.fits for simplicity.

I then ran get_dmulimits.ipynb as is and I will attach a photo of the commands and error. 
<img width="376" alt="Screen Shot 2021-09-16 at 11 13 37 am" src="https://user-images.githubusercontent.com/88965044/133533544-cb8991fa-56c2-4c0e-8951-0203bb4126fd.png">

**Comment by lpin0002** [comment]
Hey there,

I get a similar issue to this or my kernel dying, however I've instead used the same information from TCAP
![get_dmulimitsscreenshot](https://user-images.githubusercontent.com/87399967/133534335-4c43e6ef-2f1b-45ec-b60a-b4bd8b74deea.png)
I couldn't fit it in, but then I just run thistool.execute()

**Comment by sergiohcdna** [comment]
I will check again the example under `notebooks/` folder to reproduce the same error.

**Comment by kjrule** [comment]
Hi, I just made an error in my above comment. The file i created using creat_dmtable was using the following input:

python create_dmtable.py --srcname ToyDarkHalo --jfactor 5.0119e+18 --sigmav 3.0e-26 --z 0 --mpoints 100 --emin 30.0 --nebins 450

To match the data given for ToyDarkHalo

**Comment by sergiohcdna** [comment]
Ok, thanks.

**Comment by sergiohcdna** [comment]
Hi!

Sorry to keep you waiting guys.

The issue about the vector size mismatches is solved, and could find the
updated code promptly.

Cheers,
Sergio

**Comment by sergiohcdna** [comment]
The problem is because the observation container inside the `_fit_mass_point` method did not have any model.  This produce the exception when trying to compute the number of predicted events. I solved the issue by making a deep copy of the initial observation and then append the `IRFBackground` and dark-matter models to the `GModels` container. [Commit](https://github.com/sergiohcdna/ctaAnalysis/commit/171d92525c8c45376293b56ddfa30935c0472aae)

I also updated the example notebook `get_dmulimits` accordingly.
